packages/react/src/helpers/validateShippingAddressZipCode.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  type AddressBase,
+  type CountryAddressSchema,
+} from '@farfetch/blackout-client';
+
+/**
+ * Zip Code Validation Errors
+ * * ER01 - The zip code validation is ignored because is not included in the countries array to be validated
+ * * ER02 - Zip code schema not found
+ * * ER03 - The zip code inserted is invalid
+ */
+export enum ZipCodeValidationError {
+  ZipCodeValidationCountryIgnored = 'ER01',
+  ZipCodeSchemaNotFoundError = 'ER02',
+  ZipCodeInvalid = 'ER03',
+}
+
+/**
+ * Validates zip code for an array of iso-codes and returns true if the address' country code is not in the array isoCodesToValidate,
+ * or if the passed address' zip code is valid, false otherwise.
+ *
+ * @param address - address
+ * @param addressSchema - address schema to test.
+ * @param isoCodesToValidate - iso codes to be validated.
+ *
+ * @returns - boolean which is the result of the validation.
+ * */
+export default function validateShippingAddressZipCode(
+  address: AddressBase,
+  addressSchema: CountryAddressSchema[],
+  isoCodesToValidate: string[] = [],
+): { isValid: boolean; error?: string } {
+  if (!isoCodesToValidate.includes(address.country.alpha2Code)) {
+    return { isValid: true };
+  }
+
+  const schema = addressSchema.find(({ addressType }) =>
+    ['Shipping', 'Any'].includes(addressType),
```

Comment:
Actually I think this is not the same semantics as it was before. If the first addressType is `Any` and the second is `Shipping`, this will return you the `Any` address schema and not the `Shipping` one which should be given priority.

Reply:
I think this should be reverted, you are right that the schema for the `zipCode` might not change between those but I think that as a rule, the `Shipping` schema should be given priority always.

Reply:
OK, fine for me. Pls double check